xds/src/main/java/io/grpc/xds/XdsNameResolver.java

```diff
@@ -66,6 +66,8 @@
 import io.grpc.xds.XdsNameResolverProvider.CallCounterProvider;
 import io.grpc.xds.XdsNameResolverProvider.XdsClientPoolFactory;
 import io.grpc.xds.XdsRouteConfigureResource.RdsUpdate;
+import java.io.UnsupportedEncodingException;
+import java.net.URLEncoder;
```

Comment:
After offline discussions, the urlPathSegmentEscaper is slightly different than what we want as it encodes square brackets which are valid chars for authorities(IPV6 addresses).

To avoid this created another version that is similar to urlPathSegmentEscaper but also accounts for the authority specific safe chars. PTAL!